src/CoreWCF.Primitives/src/CoreWCF/Description/MessageContractHelper.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Reflection;
+using CoreWCF.Runtime;
+
+namespace CoreWCF.Description
+{
+    // goal of this class to move logic from TypeLoader to build message contract
+    static class MessageContractHelper
+    {
+
+        internal static bool IsMessageContracts(Type type)
+        {
+            foreach (Attribute attr in type.GetCustomAttributes())
+            {
+                if (String.Compare(attr.GetType().FullName, ServiceReflector.CWCFMesssageContractAttribute, true) == 0
```

Comment:
You don't need to do a string compare for the CoreWCF version of the attribute. The only reason to do the name comparison is to avoid adding System.ServiceModel.Primitives as a dependency. We have the type CoreWCF.MessageContractAttribute available.
```suggestion
                if (attr.GetType() == typeof(MesssageContractAttribute)
```

Reply:
Why not "attr is MessageContractAttribute"?